workflow/scripts/generate_virologist_output.py

```diff
@@ -30,19 +31,21 @@
 initial_reads_df = initial_reads_df[["# raw reads", "# trimmed reads"]]
 
 filtered_reads_df = pd.DataFrame()
-for file in snakemake.input.reads_filtered:
+for file in snakemake.input.reads_used_for_assembly:
     infile = open(file, "r")
-    sample = file.split("/")[-2]
-    for line in infile:
-        if "max length" in line:
-            num_reads = line.split(", ")[-2].split(" ")[0]
-            filtered_reads_df = filtered_reads_df.append(
-                {
-                    "# filtered reads": f"{int(num_reads):,}",
-                    "sample": sample,
-                },
-                ignore_index=True,
-            )
+    sample = file.split("/")[-1].split(".")[0]
```

Comment:
Add something like
```
params:
    samples=lambda wildcards: get_samples_for_date(wildcards.date)
```
to the rule, and then, change this loop to
```
for sample, file in zip(snakemake.params.samples, snakemake.input.reads_used_for_assembly)
```

Similar with other loops that need the sample name.

Reply:
added Snakemake param for samples